apple/Sources/FerrostarSwiftUI/Views/CurrentRoadNameView.swift

```diff
@@ -0,0 +1,55 @@
+import SwiftUI
+
+public struct CurrentRoadNameView: View {
+    let currentRoadName: String?
+    let theme: any RoadNameViewTheme
+    var padding: EdgeInsets = .init(top: 12, leading: 12, bottom: 12, trailing: 12)
+    var shape: AnyShape = .init(RoundedRectangle(cornerRadius: 48))
+    var borderWidth: CGFloat = 1
+
+    public init(currentRoadName: String?,
+                theme: any RoadNameViewTheme = DefaultRoadNameViewTheme())
+    {
+        self.currentRoadName = currentRoadName
+        self.theme = theme
+    }
+
+    public var body: some View {
+        if let currentRoadName {
+            Text(currentRoadName)
+                .font(theme.textFont)
+                .foregroundStyle(theme.textColor)
+                .padding(padding)
+                .background(theme.backgroundColor)
+                .clipShape(shape)
+                .overlay(shape.stroke(theme.borderColor, lineWidth: borderWidth))
+        }
+    }
+}
+
+public extension CurrentRoadNameView {
+    /// Sets the interior padding (expanding the buffer around the text).
+    func padding(_ padding: EdgeInsets) -> Self {
```

Comment:
Nice! I expect with these modifiers. We can get rid of all of the init args?

```swift
        @ViewBuilder currentRoadNameViewBuilder: @escaping (String?) -> AnyView = { name in
            AnyView(CurrentRoadNameView(currentRoadName: name, theme: DefaultRoadNameViewTheme()))
        }
```

Reply:
Ahh that's an interesting idea! Let me try that...